keepalived/vrrp/vrrp_parser.c
=======================================================
lhs: 100644 | 75353b79fabc07118df78ab9ad69c8d811f209f7
rhs: 100644 | c108de5575a4e73da092572fbe719beb9e60af7b
---@@ -120,6 +120,9 @@ vrrp_group_handler(vector_t *strvec)
 	}
 
 	vgroup->iname = read_value_block(strvec);
+
+	if (!vgroup->iname)
+		log_message(LOG_INFO, "Warning - sync group %s has empty group block", vgroup->gname);
 }
 
 static inline notify_script_t*
@@ -902,6 +905,12 @@ garp_group_interfaces_handler(vector_t *strvec)
 	garp_delay_t *gd;
 	element e;
 
+	/* Handle the interfaces block being empty */
+	if (!interface_vec) {
+		log_message(LOG_INFO, "Warning - empty garp_group interfaces block");
+		return;
+	}
+
 	/* First set the next aggregation group number */
 	delay->aggregation_group = 1;
 	for (e = LIST_HEAD(garp_delay); e; ELEMENT_NEXT(e)) {

---